lib/core/util.c

```diff
@@ -34,11 +34,14 @@ struct st_h2o_accept_data_t {
     h2o_accept_ctx_t *ctx;
     h2o_socket_t *sock;
     h2o_timeout_entry_t timeout;
-    h2o_memcached_req_t *async_resumption_get_req;
+    h2o_timeout_entry_t resumption_failure_timeout;
+    h2o_memcached_req_t *memcached_resumption_get_req;
+    h2o_redis_command_t *redis_resumption_get_command;
```

Comment:
How about creating a common structure and inheriting it for memcached and redis? By doing so, I assume you can reduce `if` statements a lot, and we would also have room to add support for other protocols without making changes to the memcached and redis bindings.

For example, something like below.

```c
struct st_h2o_accept_data_t {
    h2o_accept_ctx_t *ctx;
    h2o_socket_t *sock;
    h2o_timeout_entry_t timeout;
    struct timeval connected_at;
};

struct st_h2o_memcached_accept_data_t {
    struct st_h2o_accept_data_t super;
    h2o_memcached_req_t *memcached_resumption_get_req;
};

struct st_h2o_redis_accept_data_t {
    h2o_timeout_entry_t resumption_failure_timeout;
    h2o_redis_command_t *redis_resumption_get_command;
};
```